### DISCUSSION TITLE
Specifying callback function from Lua

### DISCUSSION THREAD

**Discussion by MichaelSenko** [neolithos/neolua#153] [Q&A]
I have a Lua program that uses LuaInterface to create a Winforms GUI.  I'm looking to move the application to 64-bit, and started running some tests with NeoLua.  I'm stuck at how to create a button and specify a Lua callback for the Click event.  In LuaInterface I do this:

local Button = luanet.import_type("System.Windows.Forms.Button")
local function callback()
	print ("click detected")
end
local okButton = Button()
okButton.Click:Add(callback)

In NeoLua, there's no problem creating the button and adding it to a form, but I haven't figured out how to specify the callback.

Thanks
Mike

**Comment by neolithos** [comment]
Hello Mike,

it is easy, but not real lua.
```
okButton.Click:Add(function (sender, e) : void 
end);
```
Only the signature must fit.

So this is also possible:

```
local function callback(sender : object, e : System.EventArgs) : void

end;

okButton.Click:Add(callback);
```
